cores/esp8266/HardwareSerial.cpp

```diff
@@ -132,6 +133,22 @@ unsigned long HardwareSerial::detectBaudrate(time_t timeoutMillis)
     return detectedBaudrate;
 }
 
+size_t HardwareSerial::readBytes(char* buffer, size_t size)
+{
+    size_t got = 0;
+
+    while (got < size)
+    {
+        esp8266::polledTimeout::oneShot timeOut(_timeout);
+        size_t avail;
+        while ((avail = available()) == 0 && !timeOut);
+        if (avail == 0)

```

Comment:
This is the correct condition. It's common to check for the counterpart, i.e.: if(timeOut) in this case, but this is better.